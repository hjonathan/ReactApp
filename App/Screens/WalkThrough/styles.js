import {StyleSheet} from 'react-native';

export default StyleSheet.create({
    slide: {
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    },
    iconContainer: {
        alignItems: 'center',
        padding: 16
    },
    icon: {
        width: 180,
        height: 180
    },
    content: {
        paddingVertical: 16,
        paddingHorizontal: 28
    },
    title: {
        color: '#FFFFFF',
        fontSize: 22,
        paddingBottom: 16,
        fontWeight: 'bold',
        textAlign: 'center'
    },
    subtitle: {
        color: '#FFFFFF',
        fontSize: 17,
        textAlign: 'center'
    }
});