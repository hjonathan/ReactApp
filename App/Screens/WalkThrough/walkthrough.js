/* eslint-disable react/no-direct-mutation-state */
import React, {Component} from "react";
import {Text, View, Image, ScrollView, Dimensions} from "react-native";
import {NavigationActions} from "react-navigation";
import Orientation from "react-native-orientation";
import {Button} from "react-native-elements";
import I18n from "../../I18n";
import Swiper from "../../Components/Swiper";
import * as Res from "../../Assets/resources";
import Styles from "./styles";

const initialState = {
    icon1: ["flex", "none", "none", "none"],
    icon2: ["flex", "none", "none"],
    icon3: ["flex", "none", "none", "none", "none"],
    icon4: ["flex", "none"],
    widthScreen: Dimensions.get("window").width
};

class WalkThrough extends Component {
    static navigationOptions= ({navigation}) => ({
        header: null
    });

    /**
     * Constructor WalkThrough
     */
    constructor () {
        super();
        this.state = initialState;
        this.onLayout = this.onLayout.bind(this);
    }

    /**
     * Charges first icon
     */
    componentDidMount () {
        Orientation.lockToPortrait();
        this.onChangeView(0);
    }

    /**
     *  Animate Icons
     *  TODO refactoring to remove the react / no-direct-mutation-state
     */
    onChangeView = (index) => {
        let count = 0;
        clearInterval(this.interval);
        this.interval = setInterval(() => {
            count += 1;
            switch (index) {
                case 0:
                    if (this.state.icon1.length > count) {
                        this.state.icon1[count - 1] = "none";
                        this.state.icon1[count] = "flex";
                        this.setState({icon1: this.state.icon1});
                    }
                    break;
                case 1:
                    if (this.state.icon2.length > count) {
                        this.state.icon2[count - 1] = "none";
                        this.state.icon2[count] = "flex";
                        this.setState({icon2: this.state.icon2});
                    }
                    break;
                case 2:
                    if (this.state.icon3.length > count) {
                        this.state.icon3[count - 1] = "none";
                        this.state.icon3[count] = "flex";
                        this.setState({icon3: this.state.icon3});
                    }
                    break;
                case 3:
                    if (this.state.icon4.length > count) {
                        this.state.icon4[count - 1] = "none";
                        this.state.icon4[count] = "flex";
                        this.setState({icon4: this.state.icon4});
                    }
                    break;
                default:
                    break;
            }
        }, 300);
    };

    /**
     * Gets current screen's width and updates withScreen state
     */
    onLayout () {
        const {width} = Dimensions.get("window");
        this.setState({widthScreen: width});
    }

    /**
     * Fires when the scroll begins to reset the icon states
     */
    onMomentumScrollBegin () {
        clearInterval(this.interval);
        this.setState({icon1: ["flex", "none", "none", "none"]});
        this.setState({icon2: ["flex", "none", "none"]});
        this.setState({icon3: ["flex", "none", "none", "none", "none"]});
        this.setState({icon4: ["flex", "none"]});
    }

    /**
     * Routes to login screen
     */
    navigate = () => {
        const routeToScreen = NavigationActions.navigate({
            routeName: "Login"
        });
        this.props.navigation.dispatch(routeToScreen);
    };

    /**
     * Renders register screen
     * @returns {*}
     */
    renderRegisterScreen () {
        return (
            <View style={{
                flex: 1,
                flexDirection: "column",
                justifyContent: "center",
                alignItems: "center",
                padding: 32,
                width: this.state.widthScreen
            }}
            >
                <View style={{margin: 16}}>
                    <Text style={Styles.subtitle}>{I18n.t("activity_walkthrough_final_page_text_1")}</Text>
                </View>
                <View style={{margin: 16}}>
                    <Button
                        raised
                        backgroundColor={Res.colors.successColor}
                        buttonStyle={{height: 38, width: 170}}
                        borderRadius={2}
                        title={I18n.t("login_btn").toUpperCase()}
                        fontSize={14}
                        fontWeight="bold"
                        onPress={() => this.navigate()}
                    />
                </View>
            </View>
        );
    }

    /**
     * Render
     * @returns {*}
     */
    render () {
        return (
            <View style={{flex: 1}} onLayout={this.onLayout}>
                <ScrollView contentContainerStyle={{flexGrow: 1}}>
                    <Swiper
                        onChangeIndex={index => this.onChangeView(index)}
                        onMomentumScrollBegin={() => this.onMomentumScrollBegin()}
                    >
                        <View style={[Styles.slide, {width: this.state.widthScreen}]}>
                            <View style={Styles.iconContainer}>
                                <Image style={[Styles.icon, {display: this.state.icon1[0]}]} source={Res.images.walkThrough.build1} />
                                <Image style={[Styles.icon, {display: this.state.icon1[1]}]} source={Res.images.walkThrough.build2} />
                                <Image style={[Styles.icon, {display: this.state.icon1[2]}]} source={Res.images.walkThrough.build3} />
                                <Image style={[Styles.icon, {display: this.state.icon1[3]}]} source={Res.images.walkThrough.build4} />
                            </View>
                            <View style={Styles.content}>
                                <Text style={Styles.title}>{I18n.t("activity_walkthrough_title_1")}</Text>
                                <Text style={Styles.subtitle}>{I18n.t("activity_walkthrough_text_1")}</Text>
                            </View>
                        </View>
                        <View style={[Styles.slide, {width: this.state.widthScreen}]}>
                            <View style={Styles.iconContainer}>
                                <Image style={[Styles.icon, {display: this.state.icon2[0]}]} source={Res.images.walkThrough.run1} />
                                <Image style={[Styles.icon, {display: this.state.icon2[1]}]} source={Res.images.walkThrough.run2} />
                                <Image style={[Styles.icon, {display: this.state.icon2[2]}]} source={Res.images.walkThrough.run3} />
                            </View>
                            <View style={Styles.content}>
                                <Text style={Styles.title}>{I18n.t("activity_walkthrough_title_2")}</Text>
                                <Text style={Styles.subtitle}>{I18n.t("activity_walkthrough_text_2")}</Text>
                            </View>
                        </View>
                        <View style={[Styles.slide, {width: this.state.widthScreen}]}>
                            <View style={Styles.iconContainer}>
                                <Image style={[Styles.icon, {display: this.state.icon3[0]}]} source={Res.images.walkThrough.report1} />
                                <Image style={[Styles.icon, {display: this.state.icon3[1]}]} source={Res.images.walkThrough.report2} />
                                <Image style={[Styles.icon, {display: this.state.icon3[2]}]} source={Res.images.walkThrough.report3} />
                                <Image style={[Styles.icon, {display: this.state.icon3[3]}]} source={Res.images.walkThrough.report4} />
                                <Image style={[Styles.icon, {display: this.state.icon3[4]}]} source={Res.images.walkThrough.report5} />
                            </View>
                            <View style={Styles.content}>
                                <Text style={Styles.title}>{I18n.t("activity_walkthrough_title_3")}</Text>
                                <Text style={Styles.subtitle}>{I18n.t("activity_walkthrough_text_3")}</Text>
                            </View>
                        </View>
                        <View style={[Styles.slide, {width: this.state.widthScreen}]}>
                            <View style={Styles.iconContainer}>
                                <Image style={[Styles.icon, {display: this.state.icon4[0]}]} source={Res.images.walkThrough.optimize1} />
                                <Image style={[Styles.icon, {display: this.state.icon4[1]}]} source={Res.images.walkThrough.optimize2} />
                            </View>
                            <View style={Styles.content}>
                                <Text style={Styles.title}>{I18n.t("activity_walkthrough_title_4")}</Text>
                                <Text style={Styles.subtitle}>{I18n.t("activity_walkthrough_text_4")}</Text>
                            </View>
                        </View>
                        {this.renderRegisterScreen()}
                    </Swiper>
                </ScrollView>
            </View>
        );
    }
}

export default WalkThrough;
