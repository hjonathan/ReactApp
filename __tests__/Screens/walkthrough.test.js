import 'react-native';
import React from 'react';
import WalkThrough from '../../App/Screens/WalkThrough';
import WalkThroughScreen from '../../App/Screens/WalkThrough/walkthrough';
import toJson from 'enzyme-to-json';
import {shallow} from "enzyme/build/index";

jest.mock('react-native-orientation', () => require.requireActual('../../__mocks__/react-native-orientation').default);

/**
 * This test is related to the render of the WalkThrough view.
 */
describe('Screen >>> Walkthrough', () => {
    const navigation = {
        dispatch: () => {}
    };
    let container, view;
    beforeEach(() => {
        container = shallow(<WalkThrough />);
        view = shallow(<WalkThroughScreen navigation={navigation}/>);
    });

    it('Render WalkThrough correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(WalkThroughScreen.navigationOptions({navigation}).header)
            .toEqual(null);
    });

    it('Render animation (first slide)', () => {
        expect(toJson(view.instance().onChangeView(0))).toMatchSnapshot();
    });
    it('Render animation (second slide)', () => {
        expect(toJson(view.instance().onChangeView(1))).toMatchSnapshot();
    });
    it('Render animation (third slide)', () => {
        expect(toJson(view.instance().onChangeView(2))).toMatchSnapshot();
    });
    it('Render animation (fourth slide)', () => {
        expect(toJson(view.instance().onChangeView(3))).toMatchSnapshot();
    });

    it('WalkThrough navigates', () => {
        let swiper = view.props('children').children.props.children.props,
            lastChildren = swiper.children.length - 1;
        expect(toJson(swiper.onChangeIndex())).toMatchSnapshot();
        expect(toJson(swiper.onMomentumScrollBegin())).toMatchSnapshot();
        expect(toJson(view.props('children').onLayout())).toMatchSnapshot();
        expect(toJson(swiper.children[lastChildren]
            .props.children[1].props.children.props.onPress())).toMatchSnapshot();
    });
});
