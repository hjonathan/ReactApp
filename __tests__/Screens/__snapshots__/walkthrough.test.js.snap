// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Walkthrough Render WalkThrough correctly 1`] = `
<Component
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <Swiper
      onChangeIndex={[Function]}
      onMomentumScrollBegin={[Function]}
    >
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "width": 750,
            },
          ]
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "padding": 16,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/build1.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "flex",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/build2.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/build3.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/build4.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
        </Component>
        <Component
          style={
            Object {
              "paddingHorizontal": 28,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "bold",
                "paddingBottom": 16,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_title_1
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_text_1
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "width": 750,
            },
          ]
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "padding": 16,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/run1.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "flex",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/run2.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/run3.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
        </Component>
        <Component
          style={
            Object {
              "paddingHorizontal": 28,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "bold",
                "paddingBottom": 16,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_title_2
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_text_2
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "width": 750,
            },
          ]
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "padding": 16,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/report1.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "flex",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/report2.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/report3.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/report4.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/report5.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
        </Component>
        <Component
          style={
            Object {
              "paddingHorizontal": 28,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "bold",
                "paddingBottom": 16,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_title_3
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_text_3
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "width": 750,
            },
          ]
        }
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "padding": 16,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/optimize1.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "flex",
                },
              ]
            }
          />
          <Image
            source={
              Object {
                "testUri": "../../../App/Assets/drawable/optimize2.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 180,
                  "width": 180,
                },
                Object {
                  "display": "none",
                },
              ]
            }
          />
        </Component>
        <Component
          style={
            Object {
              "paddingHorizontal": 28,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 22,
                "fontWeight": "bold",
                "paddingBottom": 16,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_title_4
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_text_4
          </Text>
        </Component>
      </Component>
      <Component
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "padding": 32,
            "width": 750,
          }
        }
      >
        <Component
          style={
            Object {
              "margin": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            activity_walkthrough_final_page_text_1
          </Text>
        </Component>
        <Component
          style={
            Object {
              "margin": 16,
            }
          }
        >
          <Button
            backgroundColor="#1FBC99"
            borderRadius={2}
            buttonStyle={
              Object {
                "height": 38,
                "width": 170,
              }
            }
            fontSize={14}
            fontWeight="bold"
            onPress={[Function]}
            raised={true}
            title="LOGIN_BTN"
          />
        </Component>
      </Component>
    </Swiper>
  </ScrollView>
</Component>
`;

exports[`Screen >>> Walkthrough Render animation (first slide) 1`] = `undefined`;

exports[`Screen >>> Walkthrough Render animation (fourth slide) 1`] = `undefined`;

exports[`Screen >>> Walkthrough Render animation (second slide) 1`] = `undefined`;

exports[`Screen >>> Walkthrough Render animation (third slide) 1`] = `undefined`;

exports[`Screen >>> Walkthrough WalkThrough navigates 1`] = `undefined`;

exports[`Screen >>> Walkthrough WalkThrough navigates 2`] = `undefined`;

exports[`Screen >>> Walkthrough WalkThrough navigates 3`] = `undefined`;

exports[`Screen >>> Walkthrough WalkThrough navigates 4`] = `undefined`;
